lib/screens/v2/widget_instance/subway_status.ex

```diff
@@ -468,8 +455,47 @@ defmodule Screens.V2.WidgetInstance.SubwayStatus do
       |> Enum.uniq()
       |> Enum.sort()
 
-    MapSet.size(alert_trunk_stops) > 0 or
-      alert_whole_line_stops == @green_line_branches
+    alert_whole_line_stops == @green_line_branches
+  end
+
+  # If any closed stop is served by more than one branch, the alert affects the trunk
+  defp alert_affects_gl_trunk?(
+         %Alert{
+           effect: :station_closure,
+           informed_entities: informed_entities
+         },
+         gl_stop_sets
+       ) do
+    informed_entities
+    |> Enum.map(fn e -> Map.get(e, :stop) end)
+    |> Enum.reject(&is_nil/1)
+    |> Enum.uniq()
+    |> Enum.any?(fn informed_stop ->
+      Enum.count(gl_stop_sets, &(informed_stop in &1)) > 1
+    end)
+  end
+
+  #
+  defp alert_affects_gl_trunk?(%Alert{informed_entities: informed_entities}, gl_stop_sequences) do
```

Comment:
For naming consistency with the clause above:
```suggestion
  defp alert_affects_gl_trunk?(%Alert{informed_entities: informed_entities}, gl_stop_sets) do
```
(+ update the name where it's used ofc)